### PR TITLE
chore: [PIE-4508]: update uicore & types for exec-time inputs support

### PR DESCRIPTION
**Pull request by vkbansal**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/10-common/components/MultiTypeExecutionCondition/MultiTypeExecutionCondition.tsx [hunk 1]
```diff
@@ -39,11 +39,14 @@ function MultiTypeMonacoTextFieldFixedTypeComponent(props: {
...
-function MultiTypeExecutionConditionInternal(props: MultiTypeExecutionConditionProps) {
+function MultiTypeExecutionConditionInternal(props: MultiTypeExecutionConditionProps): React.ReactElement {
   const { path, formik, allowableTypes, isInputDisabled, readonly, multiType, setMultiType, expressions } = props
 
   const conditionValue = get(formik?.values, path)
-  const allowableTypesExceptExpression = filter(allowableTypes, type => type !== MultiTypeInputType.EXPRESSION)
+  const allowableTypesExceptExpression = filter(
+    allowableTypes,
+    type => type !== MultiTypeInputType.EXPRESSION
+  ) as AllowedTypes
```
Comment (knagurski) [inline src/modules/10-common/components/MultiTypeExecutionCondition/MultiTypeExecutionCondition.tsx:49]: Any reason to not just use JS's `filter`?
```suggestion
  const allowableTypesExceptExpression = allowableTypes.filter(
    type => type !== MultiTypeInputType.EXPRESSION
  )
```

File: src/modules/10-common/components/MultiTypeFieldSelector/MultiTypeFieldSelector.tsx [hunk 2]
```diff
@@ -96,7 +96,11 @@ export function MultiTypeFieldSelector(props: ConnectedMultiTypeFieldSelectorPro
       return
     }
 
-    const runtimeValue = useExecutionTimeInput ? EXECUTION_TIME_INPUT_VALUE : RUNTIME_INPUT_VALUE
+    const runtimeValue =
+      Array.isArray(allowedTypes) &&
+      (allowedTypes as AllowedTypesWithExecutionTime[]).includes(MultiTypeInputType.EXECUTION_TIME)
```
Comment (knagurski) [inline src/modules/10-common/components/MultiTypeFieldSelector/MultiTypeFieldSelector.tsx:101]: The interface name `AllowedTypesWithExecutionTime` suggests that this is already an array, does it need `[]`?

**Review by chris-harness — APPROVED** [review]

File: src/modules/10-common/components/MultiTypeExecutionCondition/MultiTypeExecutionCondition.tsx (hunk 1 above)
Comment (vkbansal) [reply src/modules/10-common/components/MultiTypeExecutionCondition/MultiTypeExecutionCondition.tsx:49]: Not sure. This is existing code. I just updated the types.

File: src/modules/10-common/components/MultiTypeFieldSelector/MultiTypeFieldSelector.tsx (hunk 2 above)
Comment (vkbansal) [reply src/modules/10-common/components/MultiTypeFieldSelector/MultiTypeFieldSelector.tsx:101]: The interface is not an array

**Comment by vkbansal** [comment]
retrigger titlecheck

**Comment by vkbansal** [comment]
retrigger titlecheck

**Review by kapilkumawat86 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**